testutils: use non root as default for test server

e-mbrown opened the pull request:
Part of: #97535
Commit 1:
[testutils: use non root as default for test server](https://github.com/cockroachdb/cockroach/pull/101152/commits/9b8db3fabe1721f858d87dd9de39773242f190b0) 

This commit changes test from using the root
user by default to using a test user. This will
allow us to test non-root code paths and privileges.
If a test requires the root user, that can be specified
with the `requireRoot` test server args.

Commit 2:
[testutils: Fix more test failures](https://github.com/cockroachdb/cockroach/pull/101152/commits/a7f76c3c41a92dd28ecf5204cdce915f6da69656) 

This commit reverts some test to use the root user.

Release note: None

e-mbrown committed:
testutils: Fix more test failures

This commit reverts some test to use root.

Release note: None

e-mbrown committed:
testutils: Fix more test failures

This commit reverts some test to use root.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/101152)
<!-- Reviewable:end -->

e-mbrown committed:
testutils: use non root as default for test server

This commit changes test from using the root
user by default to using a test user. This will
allow us to test non-root code paths and privileges.
If a test requires the root user, that can be specified
with the `requireRoot` test server args.

Release note: None
